File: samples/profile/competitors/python/hello.py
#!/usr/bin/env python3
# Python "hello-metric" runner — emits one schema-conformant CSV row (columns.txt
# order). Real per-area Python runners (json/orjson/msgspec, numpy) extend this
# contract in Units 5+.
import os
import time

run_id = os.environ.get("PROFILE_RUN_ID", "local")
ts = os.environ.get("PROFILE_RUN_TS", "")
ver = os.environ.get("PROFILE_LANG_VERSION", "")

t0 = time.perf_counter_ns()
acc = 0
for i in range(1_000_000):
    acc += i
ns = time.perf_counter_ns() - t0
if acc < 0:
    print("", end="")

print(
    f"1,{run_id},{ts},hello,scaffold,,-1,,1000000,python,{ver},stdlib,,-OO,0,1,"
    f"{ns},{ns},{ns},{ns},,,-1,-1,-1,-1,-1,ok,true,,"
)

File: samples/profile/competitors/rust/hello.rs
// Rust "hello-metric" runner — emits one schema-conformant CSV row (columns.txt
// order). Times a trivial loop and reports it as the timing distribution. The
// real per-area Rust runners (Cargo projects with serde_json/hashbrown/etc.)
// extend this contract in Units 5+.
use std::time::Instant;

fn main() {
    let run_id = std::env::var("PROFILE_RUN_ID").unwrap_or_else(|_| "local".to_string());
    let ts = std::env::var("PROFILE_RUN_TS").unwrap_or_default();
    let ver = std::env::var("PROFILE_LANG_VERSION").unwrap_or_default();

    let t0 = Instant::now();
    let mut acc: u64 = 0;
    for i in 0..1_000_000u64 {
        acc = acc.wrapping_add(i);
    }
    let ns = t0.elapsed().as_nanos();
    std::hint::black_box(acc);

    println!(
        "1,{run_id},{ts},hello,scaffold,,-1,,1000000,rust,{ver},stdlib,,-O,0,1,{ns},{ns},{ns},{ns},,,-1,-1,-1,-1,-1,ok,true,,"
    );
}
